grammar.js

```diff
@@ -722,7 +722,7 @@ module.exports = grammar({
     )),
 
     identifier: $ => /[\a_]\w*/,
-    generic_identifier: $ => /[A-Z+]\w*\[\w*\]/,
+    generic_identifier: $ => /[A-Z+]\w*\[[,\s\w]*\]/,
```

Comment:
I think that rather than matching all of this as one `(generic_identifier)` token, we should parse it as something like `(generic_type (identifier) (identifier) (identifier))`. This way we could distinguish the name of the generic type from its type parameters based on the syntax tree.